backoff/backoff.go

```diff
@@ -0,0 +1,71 @@
+package backoff
+
+import (
+	"time"
+)
+
+type RandomGenerator interface {
```

Comment:
We should add documentation and example for the random generator from the standard library.

```
rand.NewSource(time.Now().UnixNano())
```

Reply:
Yes, I forgot to add comments. Added them in the latest commit.